features/upgrade/storage/upgrade.feature

```diff
@@ -172,3 +172,141 @@ Feature: Storage upgrade tests
+    When I run the :apply client command with:
+      | f | csi-hostpath-plugin.yaml |
+    Then the step should succeed
+
+    Given I obtain test data file "storage/csi/csi-hostpath-provisioner.yaml"
+    When I run the :apply client command with:
+      | f | csi-hostpath-provisioner.yaml |
+    Then the step should succeed
+
+    Given I obtain test data file "storage/csi/csi-hostpath-snapshotter.yaml"
+    When I run the :apply client command with:
+      | f | csi-hostpath-snapshotter.yaml |
+    Then the step should succeed
+
+    And I wait up to 360 seconds for the steps to pass:
+    """
+    Given the pod named "csi-hostpath-attacher-0" is ready
+    Given the pod named "csi-hostpath-provisioner-0" is ready
+    Given the pod named "csi-hostpath-snapshotter-0" is ready
+    Given the pod named "csi-hostpathplugin-0" is ready
+    """
+    
+    #Create storageclass volumesnapshotclass
+    Given I obtain test data file "storage/csi/csi-storageclass.yaml"
+    When I run the :apply client command with:
+      | f | csi-storageclass.yaml |
+    Then the step should succeed
+
+    Given I obtain test data file "storage/csi/snapshotclass.yaml"
+    When I run the :apply client command with:
+      | f | snapshotclass.yaml |
+    Then the step should succeed
+
+    #Create pvc/pod/volumesnapshot
+    Given I obtain test data file "storage/misc/pvc-with-storageClassName.json"
+    When I run oc create over "pvc-with-storageClassName.json" replacing paths:
+      | ["metadata"]["name"]         | pvc-hostpath    |
+      | ["spec"]["storageClassName"] | csi-hostpath-sc |
+
+    Given I obtain test data file "storage/misc/pod.yaml"
+    When I run oc create over "pod.yaml" replacing paths:
+      | ["metadata"]["name"]                                         | mypod         |
+      | ["spec"]["volumes"][0]["persistentVolumeClaim"]["claimName"] | pvc-hostpath  |
+      | ["spec"]["containers"][0]["volumeMounts"][0]["mountPath"]    | /mnt/hostpath |
+    Then the step should succeed
+    Given the pod named "mypod" becomes ready
+
+    When I execute on the pod:
+      | touch | /mnt/hostpath/test-before-upgrade |
+    Then the step should succeed
+
+    Then I execute on the pod:
+      | sync |
+    Then the step should succeed
+
+    #Create volumesnapshot
+    Given I obtain test data file "storage/csi/volumesnapshot.yaml"
+    When I run oc create over "volumesnapshot.yaml" replacing paths:
+      | ["metadata"]["name"]                            | pvc-hostpath-snapshot  |
+      | ["spec"]["volumeSnapshotClassName"]             | csi-hostpath-snapclass |
+      | ["spec"]["source"]["persistentVolumeClaimName"] | pvc-hostpath           |
+    Then the step should succeed
+    And I wait up to 180 seconds for the steps to pass:
+    """
+    When I run the :describe client command with:
+      | resource | volumesnapshot        |
+      | name     | pvc-hostpath-snapshot |
+    Then the output should match "Ready To Use\:\s+true"
+    """
+
```

Comment:
Should we verify snapshot works before the upgrade. (crating pvc  and deployment)

Reply:
I just focus on upgrade here. Suppose snapshot function will be tested in the daily run

Reply:
ok